### PR TITLE
Fix DB connection leak

### PR DESCRIPTION
**Pull request by martonarbocz**
https://kaligo.slack.com/archives/C01LK7XLGE9/p1704682961528359
We discovered that using Sequel.connect to return the connection actually leaks it. Let's use a block instead.

+ sqlite depends on pkg-config, so I added that to bin/setup

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -3,6 +3,8 @@ set -euo pipefail
 IFS=$'\n\t'
 set -vx
 
+brew install pkg-config
```
Comment (Vuta) [inline bin/setup:6]: which error did you encounter without this line?

**Review by Vuta — COMMENTED** [review]
CI is failing, can you help fix it

File: bin/setup (hunk 1 above)
Comment (martonarbocz) [reply bin/setup:6]: "pkg-config is missing" 😄

**Comment by martonarbocz** [comment]
> CI is failing, can you help fix it

Yeah I'm looking at it, quite weird because originally tests passed for me with the system Ruby, but they fail when 3.1.2.

File: lib/sequel_data/migrate/migrator.rb [hunk 2]
```diff
@@ -63,10 +65,10 @@ def table
         :data_migrations
       end
 
-      def connect_database
+      def connect_to_db(&block)
```
Comment (martonarbocz) [inline lib/sequel_data/migrate/migrator.rb]: renamed because there was a name collision in Ruby 3.0+

Comment (hieuk09) [reply lib/sequel_data/migrate/migrator.rb]: I don't think ruby 3 introduce a `connect_database` kernel method. Could you share the error you get?

**Review by hieuk09 — COMMENTED** [review]
Overall LGTM, could you add a test?

File: bin/setup (hunk 1 above)
Comment (hieuk09) [reply bin/setup:6]: I think this could happen if we install ruby without rvm or rbenv

File: lib/sequel_data/migrate/migrator.rb (hunk 2 above)
Comment (martonarbocz) [reply lib/sequel_data/migrate/migrator.rb]: The error is:

```
RBS::Test::Tester::TypeError:
       TypeError: [SequelData::Migrate::Migrator#connect_database] UnexpectedBlockError: unexpected block is given for `() -> untyped`
```

Stepping in the code I end up in the following function:


Edit: damn it, image upload is broken again...

Comment (martonarbocz) [reply lib/sequel_data/migrate/migrator.rb]: https://i.imgur.com/pV6bMFQ.png

Comment (martonarbocz) [reply lib/sequel_data/migrate/migrator.rb]: hook.rb is in the `rbs` gem

Comment (hieuk09) [reply lib/sequel_data/migrate/migrator.rb]: I pushed a fix for this error

**Comment by fill32** [comment]
how do we carry out the test? trigger the worker and see if there's any `idle` connection?

**Comment by martonarbocz** [comment]
I wanted to just do `expect ... not_change db.pool.size`, but it seems that db connections stay constant during the test run, even when I deliberately created extra connections by not using a block 🤔

**Comment by martonarbocz** [comment]
Maybe this is because of the simpler DB setup (just using SQLite)? Hieu, do you have any advice how to carry out the test reliably?

**Comment by Vuta** [comment]
@martonarbocz you can try assert the value of `Sequel::DATABASES` before and after `migrator.migrate` is called

**Comment by martonarbocz** [comment]
> @martonarbocz you can try assert the value of `Sequel::DATABASES` before and after `migrator.migrate` is called

Thank you sir, that works 💯

**Review by Vuta — APPROVED** [review]

File: spec/sequel_data/migrate/migrator_spec.rb [hunk 3]
```diff
@@ -115,6 +117,24 @@
         }.from(["20010101010205_second_migration.rb"]).to(versions)
       end
     end
+
+    context "migrations are applied" do
+      before do
+        db.create_table(:data_migrations) do
+          String :version, null: false, primary: true
+        end
+      end
```
Comment (hieuk09) [inline spec/sequel_data/migrate/migrator_spec.rb]: ~We don't need to create the table because the migrator will create the table if it does not exist.~

**Pull request merged**